misc/src/main/java/com/redhat/lightblue/savedsearch/SavedSearchCache.java

```diff
@@ -0,0 +1,388 @@
+     * versions. It returns null or empty array if nothing is found.
+     * In case of retrieval error, a RetrievalError is thrown
+     * containing the errors.
+     */
+    public JsonNode retrieveFromDB(Mediator m,
+                                   ClientIdentification clid,
+                                   String searchName,
+                                   String entity,
+                                   String version) {
+        
+        FindRequest findRequest=new FindRequest();
+        findRequest.setEntityVersion(new EntityVersion(savedSearchEntity,savedSearchVersion));
+        findRequest.setClientId(clid);
+        List<Value> versionList=new ArrayList<>(2);
+        versionList.add(NULL_VALUE);
+        // Include all segments of the version in the search list
+        if(version!=null) {
+            int index=0;
+            while((index=version.indexOf('.',index))!=-1) {
+                versionList.add(new Value(version.substring(0,index)));
+                index++;
+            }
+            versionList.add(new Value(version));
+        }
+        QueryExpression q=new NaryLogicalExpression(NaryLogicalOperator._and,
+                                                    new ValueComparisonExpression(P_NAME,
+                                                                                  BinaryComparisonOperator._eq,
+                                                                                  new Value(searchName)),
+                                                    new ValueComparisonExpression(P_ENTITY,
+                                                                                  BinaryComparisonOperator._eq,
+                                                                                  new Value(entity)),
+                                                    new ArrayContainsExpression(P_VERSIONS,
+                                                                                ContainsOperator._any,
+                                                                                versionList));
+        LOGGER.debug("Searching {}",q);
+        findRequest.setQuery(q);
+        findRequest.setProjection(FieldProjection.ALL);
+        Response response=m.find(findRequest);
+        if(response.getErrors()!=null&&!response.getErrors().isEmpty())
+            throw new RetrievalError(response.getErrors());
+        LOGGER.debug("Found {}",response.getEntityData());
+        return response.getEntityData();
+    }
+
+    /**
+     * Either loads the saved search from the db, or from the
+     * cache. 
+     *
+     * @param m Mediator instance
+     * @param clid The client id
+     * @param searchName name of the saved search
+     * @param entit Name of the entity
+     * @param version Entity version the search should run on
+     *
+     * @return The saved search document, or null if not found
+     */
+    public JsonNode getSavedSearch(Mediator m,
```

Comment:
Can we name these a little more intuitively?  `getSavedSearch` and `getSavedSearchFromDb` or something like that.